### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Towards https://github.com/giantswarm/giantswarm/issues/2328  

This PRs add initial boilerplate / ci for microclient, which is a package that provides helper functions for http clients, particularly aimed towards adding metrics to our http clients.

### CODE REVIEW

File: microclient.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// microclient provides opinionated helpers for REST clients.
+package microclient
+
+import (
+	"github.com/go-resty/resty"
+)
+
+// Do takes a resty request function and executes it, adding metrics.
+func Do(requestFunc func(string) (resty.Response, error)) (resty.Response, error) {
```
Comment (JosephSalisbury) [inline microclient.go]: Added a stub, this will be filled / changed / removed / obliterated in the next PR.

File: microclient.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+package microclient
+
+import (
+	"context"
+
+	"github.com/go-resty/resty"
+)
+
+// Do takes a resty request function and executes it, adding metrics.
+func Do(ctx context.Context, requestFunc func(string) (resty.Response, error)) (resty.Response, error) {
```
Comment (JosephSalisbury) [inline microclient.go]: Just added the stub for now, this will be filled / changed / removed / obliterated in the next PR.

File: microclient.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+	// TODO: add metrics.
+	// TODO: actually perform the request.
+	return resty.Response{}, nil
+}
```
Comment (kopiczko) [inline microclient.go]: I'm not a fan of writing wrappers over wrappers. But when it's helpful why not preserve the APIs to make life easier. This repo could be named `resty` and add define `R` func https://godoc.org/github.com/go-resty/resty#R as in the resty API. Then using the package would be just a matter of switching import and the API stays familiar. AFAIK this is a common practice.

Comment (xh3b4sd) [reply microclient.go]: We would basically need to rewrite the whole resty package. There are no interfaces. We would need to wrap everything on our own which is what you do not like. Joe and I talked about this and couldn't come up with a better idea.

Comment (kopiczko) [reply microclient.go]: > We would need to wrap everything on our own which is what you do not like

I would like that actually. We don't have to wrap everything. Just the parts we need.

**Pull request merged**